packages/react/src/utils.ts

```diff
@@ -418,3 +418,8 @@ export const getModelManager = (
 ): AnyModelManager | undefined => {
   return get(apiClient, namespaceDataPath([apiIdentifier], namespace).join("."));
 };
+
+export const getPropsWithoutRef = <T extends { ref: RefCallback<any> | RefObject<any> }>(props: T) => {
```

Comment:
Maybe add a comment here explaining polaris does not work with refs and they they need to as an example as to why the ref value may need to be removed